backvendor/vendored.go

```diff
@@ -215,15 +215,20 @@ func (src GoSource) DescribeProject(project *vcs.RepoRoot, root string) (*Refere
 	}
 	defer wt.Close()
 
-	hashes, err := NewFileHashes(wt.VCS.Cmd, root, src.excludes)
+	excludes := make(map[string]bool)
+	for key := range src.excludes {
```

Comment:
Since `src.excludes` is a map with boolean values, it'd make sense to make sure the value is `true`. It might be worth reconsidering your type for `src.excludes` in general. Perhaps you could just make this a slice instead.

Reply:
If it's really just a set that's required because a key's value is never false then a map of empty struct is the idiomatic Go way. bools take bytes, an empty struct doesn't. k := make(map[string]struct{}); k[v] = struct{}{}

Reply:
I went backwards and forwards on this. I'd [previously](adfa462b1eeadb478ed01e82aeb87254b635f4a3) used `map[string]struct{}`, yes, but then I came across a [remark in Effective Go](https://golang.org/doc/effective_go.html#maps) ("A set can be implemented as [...]") recommending `map[string]bool` instead.

Although the values now use memory, this data type has the advantage of making the look-up code a little easier to read because of the zero-value for bool being `false`:
```
if set[element] { ... }
```
vs
```
if _, ok := set[element]; ok { ... }
```

Now that I look at it again, the `map[string]struct{}` look-up is easier to read than I remembered.

Reply:
I agree with Matt that if map[string]bool then possibility of v=false has to be considered as a future change might do that rather than delete k.  For a set, the lookup could deviate from the normal `ok`, since it's the main purpose of the indexing, and be an identifier specific and explanatory to the usage:

    if _, skip := excludes[path]; skip {

Reply:
Good idea.

I think [this](https://github.com/release-engineering/backvendor/pull/29/commits/8829b46cb4f0945354b8a3f659323f9eb4784934#diff-8436dcf86d74d5ebc5f7fc910e62996fR171) was the slightly awkward look-up that made me think `bool` was an easier choice: combining the membership test with some other expression.

Anyway, updated. I split out the datatype change into its own commit.